spec/integration/strong_resource_spec.rb

```diff
@@ -209,6 +209,38 @@
       end
     end
 
+      context "jsontype of subclasses" do
+        let(:params) do
+          { unicorn: { title: 'Jet.com' }, data: { type: 'unicorns' } }
+        end
+
+        it 'should raise exception' do
+          expect {
+            controller.strong_resource.to_h
+          }.to raise_error StrongResources::UnregisteredResource
+        end
+      end
+  end
+
+  context "in subclasses" do
+    let(:controller) do
+      controller = UnicornCompaniesController.new
```

Comment:
once again, I'd really prefer this in the test setup itself using Class.new to be more expressive (that this is a subclass) and avoid bleeding into other tests.